tests/test_examples.py

```diff
@@ -0,0 +1,227 @@
+                          ('widgets/image_view.enaml', None),
+                          ('widgets/main_window.enaml', None),
+                          ('widgets/menu_bar.enaml', None),
+                          pytest.param('widgets/mpl_canvas.enaml', None,
+                                       marks=pytest.mark.skipif(
+                                           not MATPLOTLIB_AVAILABLE,
+                                           reason='Requires matplotlib')),
+                          ('widgets/notebook.enaml', None),
+                          ('widgets/popup_menu.enaml', None),
+                          ('widgets/popup_view.enaml',
+                           handle_popup_view_example),  # Requires handler
+                          ('widgets/progress_bar.enaml', None),
+                          ('widgets/scroll_area.enaml', None),
+                          ('widgets/slider.enaml', None),
+                          ('widgets/spin_box.enaml', None),
+                          ('widgets/splitter.enaml', None),
+                          ('widgets/tool_bar.enaml', None),
+                          ('widgets/tool_buttons.enaml', None),
+                          ('widgets/v_group.enaml', None),
+                          pytest.param('widgets/vtk_canvas.enaml', None,
+                                       marks=pytest.mark.skipif(
+                                           not VTK_AVAILABLE,
+                                           reason='Requires vtk')),
+                          ('widgets/window_children.enaml', None),
+                          ('widgets/window_closing.enaml',
+                           handle_window_closing),
+                          ('widgets/window.enaml', None)])
+def test_examples(qt_app, enaml_sleep, path, handler):
+    """ Test the enaml examples.
+
+    """
+    dir_path = os.path.abspath(os.path.split(os.path.dirname(__file__))[0])
+    enaml_file = os.path.join(dir_path, 'examples', os.path.normpath(path))
+
+    with open(enaml_file, 'rU') as f:
+        enaml_code = f.read()
+
+    # Parse and compile the Enaml source into a code object
+    ast = parse(enaml_code, filename=enaml_file)
+    code = EnamlCompiler.compile(ast, enaml_file)
+
+    # Create a proper module in which to execute the compiled code so
+    # that exceptions get reported with better meaning
+    try:
+        module = types.ModuleType('enaml_test')
+        module.__file__ = os.path.abspath(enaml_file)
+        sys.modules['enaml_test'] = module
+        ns = module.__dict__
+
+        # Put the directory of the Enaml file first in the path so relative
+        # imports can work.
+        sys.path.insert(0, os.path.abspath(os.path.dirname(enaml_file)))
+        with imports():
+            exec code in ns
+
+        with close_all_windows(qt_app):
+            window = ns['Main']()
+            window.show()
+            window.send_to_front()
+            qt_app.process_events()
+            sleep(enaml_sleep)
```

Comment:
This is not working on my system in some cases. I created a simple hello world example with a window and a label, the text doesn't display because it the sleep is blocking. 

I had to change it to:

```python

            while enaml_sleep>0:
                qt_app.process_events()
                sleep(0.1)
                enaml_sleep -= 0.1

```

I don't think using a blocking sleep is ever a good idea

Reply:
This is indeed not working perfectly well. I hope to rework this soon for a better solution. I think that pytest-qt may help here.